### DISCUSSION TITLE
'Invalid or incomplete response' when using Mistral API

### DISCUSSION THREAD

**Discussion by Adamchanadam** [crewAIInc/crewAI#1143] [Q&A]
I have spent lot of time to understanding the Mistral API within the CrewAI framework, specifically for 'mistral-medium' and 'mistral-small'. Although I am able to execute the code smoothly, I am encountering an issue: it returns the error 'Invalid or incomplete response. Parsing LLM output produced both a final answer and a parse-able action.'

To address this, I have attempted the following solutions, but without success:

Updating the versions of the relevant packages: langchain to 0.1.1, langchain_mistralai to 0.0.3, and crewai to 0.1.32.
Adjusting the parameters such as temperature, max_tokens, and top_p across various ranges.

Does anyone have a solution for this problem?"
Thanks

**Comment by Adamchanadam** [comment]
I've wrote email to Mistral AI and the reply is they don't support Stop parameter at this moment. They're still working on it.

**Comment by j-dominguez9** [comment]
Also having an issue with this.

**Comment by joaomdmoura** [comment]
can you try the new RC and tell me if it works better?
`pip install crewai==0.17.0rc2`
--
João Moura
@joaomdmoura <http://twitter.com/joaomdmoura>


Em sáb., 2 de mar. de 2024 às 17:05, Joaquin Dominguez <
***@***.***> escreveu:

**Comment by j-dominguez9** [comment]
I just tried it with the new rc you specified and still getting same error:

![image](https://github.com/joaomdmoura/crewAI/assets/83036592/1ec95ba6-e2a6-4e49-9165-4510b3fd7fd2)
![image](https://github.com/joaomdmoura/crewAI/assets/83036592/abea5594-6f9d-46d2-8130-414a3043ecfb)
